src/cpp/rtps/writer/StatefulWriter.cpp

```diff
@@ -558,16 +572,18 @@ bool StatefulWriter::matched_reader_add(RemoteReaderAttributes& rdata)
         assert(last_seq + 1 == current_seq);
 
         std::vector<GUID_t> guids(1, rp->m_att.guid);
-        const LocatorList_t& locatorsList = rp->m_att.endpoint.remoteLocatorList;
-        RTPSMessageGroup group(mp_RTPSParticipant, this, RTPSMessageGroup::WRITER, m_cdrmessages, locatorsList, guids);
+        const LocatorList_t remote_locators_shrinked{mp_RTPSParticipant->network_factory().ShrinkLocatorLists(
+                {rp->m_att.endpoint.remoteLocatorList})};
```

Comment:
If I remeber correctly, remoteLocatorList is shrinked inside `matched_reader_add`, so there is no need to shrink it here

Reply:
Decided by the team to keep in this way.